patches/server/1055-Deprecate-ItemStack-setType.patch

```diff
@@ -0,0 +1,26 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Tue, 26 Mar 2024 21:42:23 -0400
+Subject: [PATCH] Deprecate ItemStack#setType
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+index f67b048c3831fcab24acd12249cbd32d386c3ffe..e333ea94e3c495abe073d94c45816a239bc3bc6e 100644
+--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
++++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+@@ -711,4 +711,15 @@ public final class CraftItemStack extends ItemStack {
+     static boolean hasItemMeta(net.minecraft.world.item.ItemStack item) {
+         return !(item == null || item.getTag() == null || item.getTag().isEmpty());
+     }
++    // Paper start - with type
++    @Override
++    public ItemStack withType(final Material type) {
++        ItemStack itemStack = asNewCraftStack(CraftItemType.bukkitToMinecraft(type), this.getAmount());
```

Comment:
This goes into a double conversion (Material -> Item (internal) -> Material) for no reason? Also you can set the item meta directly using the regular constructor.